src/string_immutable.zig

```diff
@@ -5816,49 +5824,31 @@ pub fn visibleCodepointWidthType(comptime T: type, cp: T) usize {
 }
 
 pub const visible = struct {
-    fn visibleASCIIWidth(input_: anytype) usize {
+    // Ref: https://cs.stanford.edu/people/miles/iso8859.html
+    fn visibleLatin1Width(input: []const u8) usize {
         var length: usize = 0;
-        var input = input_;
-
-        if (comptime Environment.enableSIMD) {
-            // https://zig.godbolt.org/z/hxhjncvq7
-            const ElementType = std.meta.Child(@TypeOf(input_));
-            const simd = 16 / @sizeOf(ElementType);
-            if (input.len >= simd) {
-                const input_end = input.ptr + input.len - (input.len % simd);
-                while (input.ptr != input_end) {
-                    const chunk: @Vector(simd, ElementType) = input[0..simd].*;
-                    input = input[simd..];
-
-                    const cmp: @Vector(simd, ElementType) = @splat(0x1f);
-                    const match1: @Vector(simd, u1) = @bitCast(chunk >= cmp);
-                    const match: @Vector(simd, ElementType) = match1;
-
-                    length += @reduce(.Add, match);
-                }
-            }
-
-            // this is a deliberate compiler optimization
-            // it disables auto-vectorizing the "input" for loop.
-            if (!(input.len < simd)) unreachable;
-        }
-
-        for (input) |c| {
-            length += if (c > 0x1f) 1 else 0;
-        }
-
+        for (input) |c| length += visibleLatin1WidthScalar(c);
```

Comment:
This should use `@Vector` when the input is >= 16 bytes.

Reply:
Depending on what godbolt says, this would probably work. It looks silly to paste it this way instead of using `inline for` but we want to make sure that the compiler knows the entire array is `const`

```zig
const input_end_ptr = input.ptr + input.len - (input.len % 16);
while (input.ptr != input_end_ptr) {
  const input_chunk: [16]u8 = input.ptr[0..16].*;
  const sums = [16]u8{
     visibleLatin1WidthScalar(input_chunk[0]),
     visibleLatin1WidthScalar(input_chunk[1]),
     visibleLatin1WidthScalar(input_chunk[2]),
     visibleLatin1WidthScalar(input_chunk[3]),
     visibleLatin1WidthScalar(input_chunk[4]),
     visibleLatin1WidthScalar(input_chunk[5]),
     visibleLatin1WidthScalar(input_chunk[6]),
     visibleLatin1WidthScalar(input_chunk[7]),
     visibleLatin1WidthScalar(input_chunk[8]),
     visibleLatin1WidthScalar(input_chunk[9]),
     visibleLatin1WidthScalar(input_chunk[10]),
     visibleLatin1WidthScalar(input_chunk[11]),
     visibleLatin1WidthScalar(input_chunk[12]),
     visibleLatin1WidthScalar(input_chunk[13]),
     visibleLatin1WidthScalar(input_chunk[14]),
     visibleLatin1WidthScalar(input_chunk[15]),
  };
  length += @reduce(.Add, sums); 
  input.ptr += 16;
}
```